external/fv3fit/fv3fit/pytorch/graph/train.py

```diff
@@ -119,15 +124,25 @@ def train_graph_model(
     return predictor
 
 
-def build_model(graph_network, n_state: int):
+def build_model(graph_network, n_state: int, nx: int):
     """
     Args:
         graph_network: configuration of the graph network
         n_state: number of state variables
     """
-    train_model = GraphUNet(graph_network, in_channels=n_state, out_dim=n_state).to(
-        DEVICE
-    )
+    if isinstance(graph_network, MPGUNetGraphNetworkConfig):
+        train_model = MPGraphUNet(
+            graph_network, in_channels=n_state, out_dim=n_state, nx=nx
+        ).to(DEVICE)
+    elif isinstance(graph_network, UNetGraphNetworkConfig):
+        train_model = GraphUNet(graph_network, in_channels=n_state, out_dim=n_state).to(
+            DEVICE
+        )
```

Comment:
Instead of having an `isinstance` check, make a `.build(n_state, nx)` method (you can retrieve graph_network from the `nx` argument using the helper function).